sympy/functions/elementary/exponential.py

```diff
@@ -506,7 +506,9 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             nterms = ceiling(n/cf)
         exp_series = exp(t)._taylor(t, nterms)
         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
-        if r.subs(logx, log(x)) == self:
+        if logx is None and r == self:
+            return r
+        elif r.subs(logx, log(x)) == self:
```

Comment:
I'm not sure what the correct behaviour should be here. Previously this was calling `r.subs(None, log(x))` which would just return `r` unchanged. This just leaves `r` unchanged if `logx` is None which should be consistent with the previous behaviour.